### ISSUE TITLE
eslint-import-resolver-webpack with Next.js

### ISSUE THREAD

**Issue by ChristianJacobsen** [benmosher/eslint-plugin-import#1286]
Hey!

I'm trying to get `eslint-import-resolver-webpack` working with [Next.js](https://github.com/zeit/next.js/) but my linter keeps screaming at me.

The way Next.js allows you to extend the webpack config is through their own `next.config.js`.
Like this:
```js
const path = require('path');

module.exports = {
	webpack: config => {
		// Add aliasing
		config.resolve.alias = {
			...config.resolve.alias,
			components: path.resolve('components')
		};

		return config;
	}
};
```

This successfully allows me to build the codebase, but eslint keeps reporting `import/no-unresolved`.

The `"settings"` of my `.eslintrc.json` looks like this:
```json
{
    "settings": {
        "import/resolver": {
            "webpack": {
                "config": "./next.config.js"
            }
        }
    }
}
```
If any of you have used aliased imports with Next.js (or similar apps where the `webpack.config.js` is not directly accessible) before, then I would love some help figuring this out!

**Comment by ljharb** [comment]
It seems like the webpack config is actually `require('./next.config.js').webpack` - maybe you need to put that in a module by itself?

**Comment by ChristianJacobsen** [comment]
@ljharb Thanks for that suggestion, but didn't work. The `.webpack` provided by `next.config.js` is a function taking the original config as a parameter. I tried to find the `webpack.config.js` in the Next.js package, but it's generated during the building. I guess I'll ask the devs if there's a way to expose it easier. Thanks though!

**Comment by ljharb** [comment]
In that case it seems like it’s an issue with next.js not exposing the full config; closing. Happy to reopen if it turns out there’s a change needed here. (please link to the next.js issue once you file it)

**Comment by RonnyO** [comment]
Because next.js generates the webpack config like @ChristianJacobsen mentioned, you can't import it from the next config.
So I've managed to work around this by putting the alias config in a third file, say `alias-config.js`:
```js
const path = require('path');

module.exports = {
    components: path.join(__dirname, 'components/'),
    // ...
};
````

Then using it from `webpack.config.js` to assist the eslint resolver:
```js
const alias = require('./alias-config');

module.exports = {
    resolve: {
        alias
    }
};
```

And in `next.config.js`, for the app itself:
```js
const aliases = require('./alias-config');

module.exports = {
    webpack (config) {
        const { alias } = config.resolve;
        config.resolve.alias = {
            ...alias,
            ...aliases
        };
        return config;
    }
};
```

You may need to reload your editor/webpack to see the import errors go away.

**Comment by piuccio** [comment]
I could get the previous configuration to work because of some obscure error in styled-components, so I found this configuration to work for me

Inside `.eslintc.json` or wherever you keep your eslint config

```js
"settings": {
    "import/resolver": {
      "node": {
        "moduleDirectory": [
          "node_modules",
          "app"
        ]
      }
    },
}
```

works ok if you're simply trying to convert from `../components` to `components`, won't work if you have some fancy alias configuration
